dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -63,13 +71,60 @@ def __add__(self, other: CategoricalColumn) -> CategoricalColumn:
             )
 
         merged_profile = CategoricalColumn(None)
-        merged_profile._categories = utils.add_nested_dictionaries(
-            self._categories, other._categories
-        )
         BaseColumnProfiler._add_helper(merged_profile, self, other)
-        self._merge_calculations(
-            merged_profile.__calculations, self.__calculations, other.__calculations
-        )
+
+        # If both profiles have not met stop condition
+        if not (self._stop_condition_is_met or other._stop_condition_is_met):
+            merged_profile._categories = utils.add_nested_dictionaries(
+                self._categories, other._categories
+            )
+
+            self._merge_calculations(
+                merged_profile.__calculations, self.__calculations, other.__calculations
+            )
```

Comment:
This still needs to be conducted above. right now we don't have it populated, but it should be done in either case.